src/index.js

```diff
@@ -0,0 +1,167 @@
+
+      taskElement.querySelector('.edit-manager').classList.add('hidden');
+      taskElement.querySelector('.list-editor').classList.remove('hidden');
+      printList();
+    }
+  };
+
+  // Start editting
+  document.querySelectorAll('.edit-task').forEach((etb) => {
+    etb.addEventListener('click', (e) => {
+      const taskIndex = parseInt(e.target.parentNode.parentNode.id, 10);
+      document.getElementById(`edit${taskIndex}`).value = document.querySelector(`#task${taskIndex}`).innerHTML;
+      const taskElement = document.getElementById(`${taskIndex}`);
+      const editTask = document.querySelector(`#edit${taskIndex}`);
+      editTask.classList.remove('hidden');
+      document.querySelector(`#task${taskIndex}`).classList.add('hidden');
+
+      taskElement.querySelector('.edit-manager').classList.remove('hidden');
+      taskElement.querySelector('.list-editor').classList.add('hidden');
+
+      editTask.addEventListener('keypress', (e) => {
+        const keypressed = (editTask) ? e.keyCode : e.which;
+        if (keypressed === 13) {
+          if (editTask.value !== '') { confirmEditTask(e); }
+        }
+      });
+    });
+  });
+
+  // Confirm task edition
+  document.querySelectorAll('.edit-confirm').forEach((ec) => {
+    ec.addEventListener('click', confirmEditTask);
+  });
+
+  // Cancel task etition
+  document.querySelectorAll('.edit-cancel').forEach((ec) => {
+    ec.addEventListener('click', (e) => {
+      const taskIndex = parseInt(e.target.parentNode.parentNode.id, 10);
+      const taskElement = document.getElementById(`${taskIndex}`);
+
+      document.querySelector(`#edit${taskIndex}`).classList.add('hidden');
+      document.querySelector(`#task${taskIndex}`).classList.remove('hidden');
+
+      taskElement.querySelector('.edit-manager').classList.add('hidden');
+      taskElement.querySelector('.list-editor').classList.remove('hidden');
+
+      printList();
+    });
+  });
+
+  // delete selected task
+  document.querySelectorAll('.delete-task').forEach((dtb) => {
+    dtb.addEventListener('click', (e) => {
+      const task = new Task();
+      const taskIndex = parseInt(e.target.parentNode.parentNode.id, 10);
+      task.Delete(taskIndex);
+      printList();
+    });
+  });
+
+  checkCompleted();
+  setCheckboxListener();
+};
+
+if (getToDoList() !== []) {
+  printList();
+}
+
+const addNewTask = () => {
+  if (newTaskInput.value !== '') {
+    toDoList = getToDoList();
+    const newIndex = toDoList.length + 1;
+    const task = new Task(newTaskInput.value, newIndex);
+    task.Add();
+    newTaskInput.value = '';
+    printList();
+  }
+};
+
+// Add new Task
+document.querySelector('#add').addEventListener('click', addNewTask);
+newTaskInput.addEventListener('keypress', (e) => {
+  const keypressed = (newTaskInput) ? e.keyCode : e.which;
+  if (keypressed === 13) {
+    if (newTaskInput.value !== '') { addNewTask(); }
+  }
+});
```

Comment:
You are repeating the code from 79 to 82 lines.  I believe you should create a function to call it from both places, [Learn more](https://www.taniarascia.com/refactoring-dont-repeat-yourself/).

Reply:
The code is not really repeated as the core if that takes the value is executing a different function.

Reply:
Because it becomes complex, you can keep it as it is. And minimize it by using the below code.
```suggestion
   if (e.which === 13) {
          confirmEditTask(e);
    }
});
```